### DISCUSSION TITLE
return the exact data to be sent to the client from webserver

### DISCUSSION THREAD

**Discussion by RainerZufall187** [aio-libs/aiohttp#7669] [Q&A] [answered]
i need to send the client this `b"HTTP/1.0 200 OK\r\n\r\n"` how

**Comment by Dreamsorcerer** [comment, marked as answer]
As an HTTP library, it's designed to abstract away the HTTP parts, so it's not really designed to send exact responses.

Without hacking things, the version comes from the request `request.version`, so if that's `1.0`, then you should get the correct version. The status code defaults to 200, and you can explicitly set the reason (`web.Response(reason="OK")`).

Several default headers will be added though. A hacky way to ensure none are included may be to subclass Response and then replace `_write_headers()`, removing the headers from the last line: https://github.com/aio-libs/aiohttp/blob/master/aiohttp/web_response.py#L416
Or just doing:
```
    async def _write_headers(self) -> None:
        self._headers = {}
        await super()._write_headers()
```
